bin/node-template/runtime/src/lib.rs

```diff
@@ -344,7 +344,8 @@ impl_runtime_apis! {
 
 	impl sp_api::Metadata<Block> for Runtime {
 		fn metadata() -> OpaqueMetadata {
-			Runtime::metadata().into()
+			use codec::Encode as _;
+			OpaqueMetadata::new(Runtime::metadata().encode())
```

Comment:
I'd be reluctant to add a `substrate` dependency to `frame-metadata` just for this. One goal of the separate lib is to decouple it from substrate. We could either:

- `OpaqueMetadata::new(Runtime::metadata().into())` because `RuntimeMetadataPrefixed` already implements `Into<Vec<u8>>`
- Use the equivalent `OpaqueMetadata` type from `frame-metadata`, but that would mean adding a dependency on `frame-metadata` from `sp-api` though.